tests/unit/info.tcl

```diff
@@ -0,0 +1,58 @@
+proc cmdstat {cmd} {
+    return [cmdrstat $cmd r]
+}
+
+proc errorstat {cmd} {
+    return [errorrstat $cmd r]
+}
+
+start_server {tags {"info"}} {
+    start_server {} {
+        set master_host [srv 0 host]
+        set master_port [srv 0 port]
+
+        test {commandstats and errorstats: command failed count} {
+            r config resetstat
+            assert_match {} [errorstat ERR]
+            catch {r set k} e
+            assert_match {*ERR*} $e
+            assert_match {*count=1*} [errorstat ERR]
+            assert_match {*calls=0,*,failed_calls=1} [cmdstat set]
+        }
+
+        test {errorstats: authentication error} {
+            r config resetstat
+            assert_match {} [errorstat ERR]
+            catch {r auth k} e
+            assert_match {ERR AUTH*} $e
+            assert_match {*count=1*} [errorstat ERR]
+            r config resetstat
+            assert_match {} [errorstat ERR]
+        }
+
+        test {errorstats: OOM error} {
```

Comment:
let's add a test that checks `-NOGROUP` and `-WRONGTYPE`, and an invalid command like `catch {r asdfasdf}`.
both are different classes of errors than the ones you tested.
i.e. one will have both `calls` and `failed_calls` showing `1`. and the other will show an error with nothing in the command table.

This attempt will immediately expose two facts:
1. some commands fail with an error without being executed, and others fail inside the command. maybe we want to distinguish between `failed_calls` and `rejected_calls` in the command stats? (we can do that in `call()`)

2. the problem which was discussed before, which is that many commands use a plain `addReply` to reply with an error, and we can't log these today.
the only way out that i can think of is to grep for `addReply(*,shared.*err);` and modify each of these to use a new `addErrorReply` or `addReplyErrObject`.
and even that's not enough since i see some places do funny things like:
```c
    addReplySds(c,
        sdsnew("-WRONGTYPE Key is not a valid "
               "HyperLogLog string value.\r\n"));
```
if we do this, let's do that in a separate commit, so a reviewer (now or in the future) on your new code won't have to scan changes in all other files. (i.e. we won't squash-merge it)

Reply:
@oranagra with regards to the tests I've added the requested ones and now we have:
The added tests are divided as follow:
- **failed** call authentication error
- **failed** call NOGROUP error
- **rejected** call due to wrong arity
- **rejected** call by OOM error
- **rejected** call by authorization error

With regards to 1) accounting for rejected vs failure I've added a new flag to the command when we call `rejectCommand` or `rejectCommandFormat`, meaning if we reject a command it wont increment the failed counter. 

With regards to 2) I believe @madolson has already kicked off that effort on https://github.com/redis/redis/pull/8226 so we should push 2) there correct?

I believe all comments have been addressed. Can you check if the added changes make sense? Thank you both for reviewing this and pushing this feature forward.

Reply:
sorry, i'm not able to catch up on my notifications lately, so i wasn't aware of that other PR.
i didn't review all your changes yet, but i do wee you addressed this comment.
the only things in this comment that are not addressed are:
* WRONGTYPE, which you can't address yet, until the other PR is merged, but also that's the reason i suggested the NOGROUP error, since it's already in the state today that the WRONGTYPE will be after the other PR is merged. so i guess we can skip that (or after the other PR is merged, replace the uncommon NOGROUP test with a simpler common WRONGTYPE test)
*  the unknown (asdf) command (which will leave a trace in the error stats, and not in the command stats). i think we should add it.

Reply:
@oranagra with regards to the unknown command test just added it in the latest commit.